### PR TITLE
Stream API (part II)

### PR DESCRIPTION
**Pull request by Sarmat92**
Stream api (part II)

### CODE REVIEW

File: forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Model/User.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+
+public class User {
+
```
Comment (KaydunovDenis) [inline forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Model/User.java:12]: delete empty lines

File: forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Model/UserStatus.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package JavaHomework.Lectures_17.Stream.secondPart.Model;
+
+public enum UserStatus {
+
```
Comment (KaydunovDenis) [inline forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Model/UserStatus.java:4]: empty line

File: forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Service/Service.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package JavaHomework.Lectures_17.Stream.secondPart.Service;
+
+import JavaHomework.Lectures_17.Stream.secondPart.Model.User;
+
+import java.util.List;
+
+public interface Service {
```
Comment (KaydunovDenis) [inline forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Service/Service.java]: Service -> UserService

File: forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Service/UserService.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+import JavaHomework.Lectures_17.Stream.secondPart.Model.User;
+import JavaHomework.Lectures_17.Stream.secondPart.Model.UserStatus;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static JavaHomework.Lectures_17.Stream.secondPart.Repository.UserRepository.*;
+
+public class UserService implements Service {
```
Comment (KaydunovDenis) [inline forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Service/UserService.java]: UserService -> UserServiceImpl

File: forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Service/UserService.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+    @Override
+    public void showUserList() {
+        userList
```
Comment (KaydunovDenis) [inline forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Service/UserService.java]: здесь не нужно с новой строки

File: forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Service/UserService.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+    @Override
+    public List<User> checkUserStatus() {
+        return userList.stream()
+                .filter(it -> it.getUserStatus().equals(UserStatus.GUEST))
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public void addUserList() {
```
Comment (KaydunovDenis) [inline forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Service/UserService.java]: addUserList -> initialUserList

File: forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/App.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
+package JavaHomework.Lectures_17.Stream.secondPart;
+
+import JavaHomework.Lectures_17.Stream.secondPart.Service.UserService;
+
+
+public class App {
+
```
Comment (KaydunovDenis) [inline forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/App.java:7]: delete empty lines

**Review by KaydunovDenis — CHANGES_REQUESTED** [review]

File: forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Model/User.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    private Integer id;
+    private String name;
+    private String lastName;
+    private String nickname;
+    private Integer age;
+    private String email;
+    private Integer password;
```
Comment (AlexMackevich) [inline forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Model/User.java:19]: Perhaps, the password can include letters. This way, better use String

File: forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Model/User.java [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+
+public class User {
+
+    private Integer id;
```
Comment (AlexMackevich) [inline forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Model/User.java:13]: I created ID by long type.

File: forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Service/UserService.java [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+        userList.add(new User(2, "Dima", "Rogov", "Rogatka", 18,
+                "dimarog@yandex.ru", 123, UserStatus.USER, true));
+
+        userList.add(new User(3, "Sasha", "Gromov", "SashaGrom", 25,
+                "gromov@mail.ru", 852, UserStatus.VIP_USER, false));
+
+        userList.add(new User(4,"Karolina","Kas","Karolinka",17,
+                "Kasik@gmail.com", 1405,UserStatus.USER,false));
+    }
+}
```
Comment (AlexMackevich) [inline forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Service/UserService.java]: IDK, correctly, what you include initialization like this way. In my opinion, it's better to keep it in a repository

File: forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Service/UserService.java [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+         return userList.stream()
+                .peek(it->it.setUserStatus(UserStatus.USER))
+                 .collect(Collectors.toList());
+    }
+
+    @Override
+    public List<User> checkUserStatus() {
+        return userList.stream()
+                .filter(it -> it.getUserStatus().equals(UserStatus.GUEST))
+                .collect(Collectors.toList());
```
Comment (AlexMackevich) [inline forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Service/UserService.java]: As I see, you took Users, found who has status GUEST but didn't take the filtered list. If you have the filtered list with status GUEST, why you didn't use it in the method changeUserStatusGuestToUser?

If you don't take the filtred list, for what you filtred them?

File: forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Service/Service.java [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+
+    List<User> checkUserStatus();
+
+    void addUserList();
```
Comment (AlexMackevich) [inline forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Service/Service.java]: I don't think the interface should store this

File: forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Service/Service.java [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+
+    List<User> changeUserStatusGuestToUser();
+
+    List<User> changeIsActive();
+
+    void showUserList();
```
Comment (AlexMackevich) [inline forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Service/Service.java]: And like this

File: forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Model/User.java (hunk 9 above)
Comment (KaydunovDenis) [reply forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Model/User.java:13]: Согласен, общепринятая практика для id создавать long типа

Comment (Sarmat92) [reply forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Model/User.java:13]: done

File: forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Model/User.java (hunk 8 above)
Comment (Sarmat92) [reply forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Model/User.java:19]: done

File: forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Service/UserService.java (hunk 6 above)
Comment (Sarmat92) [reply forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Service/UserService.java]: done

File: forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Service/UserService.java (hunk 4 above)
Comment (Sarmat92) [reply forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Service/UserService.java]: done

File: forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Service/Service.java (hunk 3 above)
Comment (Sarmat92) [reply forJIS7/src/main/java/JavaHomework/Lectures_17.Stream/secondPart/Service/Service.java]: done

**Review by KaydunovDenis — APPROVED** [review]

**Review by ArtikTico — APPROVED** [review]
GJ

**Pull request merged**